akka-testkit-typed/src/main/scala/akka/testkit/typed/TestKit.scala

```diff
@@ -1,51 +1,106 @@
+          system.printTree)
+        if (verifySystemShutdown) throw new RuntimeException(msg)
+        else println(msg)
+    }
+  }
+
+}
+
 /**
  * Testkit for typed actors. Extending this removes some boiler plate when testing
  * typed actors.
  *
  * If a test can't extend then use the [[TestKitBase]] trait
- *
- * @param _system The [ActorSystem] for the test
  */
 @ApiMayChange
-class TestKit(_system: ActorSystem[_]) extends TestKitBase {
-  implicit val system = _system
+class TestKit(name: String, config: Option[Config]) extends TestKitBase {
+  def this() = this(TestKit.getCallerName(classOf[TestKit]), None)
+  def this(name: String) = this(name, None)
+  def this(config: Config) = this(TestKit.getCallerName(classOf[TestKit]), Some(config))
+  def this(name: String, config: Config) = this(name, Some(config))
+  import TestKit._
+  implicit val system = ActorSystem(testKitGuardian, name, config = config)
 }
 
 @ApiMayChange
 trait TestKitBase {
-  def system: ActorSystem[_]
+  def system: ActorSystem[TestKitCommand]
   implicit def testkitSettings = TestKitSettings(system)
+  implicit def scheduler = system.scheduler
+  private val childName: Iterator[String] = Iterator.from(0).map(_.toString)
+  // FIXME testkit config
+  private val timeoutDuration = 5.seconds
+  implicit private val timeout = Timeout(timeoutDuration)
 
   def shutdown(): Unit = {
-    shutdown(system, 5.seconds)
+    TestKit.shutdown(system, timeoutDuration)
   }
 
-  def shutdown(
-    actorSystem:          ActorSystem[_],
-    duration:             Duration,
-    verifySystemShutdown: Boolean        = false): Unit = {
-    system.terminate()
-    try Await.ready(actorSystem.whenTerminated, duration) catch {
-      case _: TimeoutException ⇒
-        val msg = "Failed to stop [%s] within [%s] \n%s".format(actorSystem.name, duration,
-          actorSystem.printTree)
-        if (verifySystemShutdown) throw new RuntimeException(msg)
-        else println(msg)
-    }
-  }
+  /**
+   * Spawn the given behavior. This is created as a child of the test kit
+   * guardian
+   */
+  def spawn[T](behavior: Behavior[T]): ActorRef[T] =
+    Await.result(system ? (SpawnActorAnonymous(behavior, _)), timeoutDuration)
+
+  /**
+   * Spawn the given behavior. This is created as a child of the test kit
+   * guardian
+   */
+  def spawn[T](behavior: Behavior[T], name: String): ActorRef[T] =
+    Await.result(system ? (SpawnActor(name, behavior, _)), timeoutDuration)
 
   // The only current impl of a typed actor system returns a Future.successful currently
   // hence the hardcoded timeouts
-  def actorOf[T](behaviour: Behavior[T], name: String): ActorRef[T] =
-    Await.result(system.systemActorOf(behaviour, name)(Timeout(20.seconds)), 21.seconds)
-}
+  def systemActor[T](behaviour: Behavior[T], name: String): ActorRef[T] =
+    Await.result(system.systemActorOf(behaviour, name), timeoutDuration)
```

Comment:
Remove the comment above that talks about the now-removed hardcoded timeouts

Reply:
Done